hangman.py

```diff
@@ -0,0 +1,250 @@
+    returns: boolean - True and append letter to list available letters,
+    otherwise False, print information message about
+    repeated  letter or use of a banned symbol
+    and take away warnings;
+    """
+    global letters, letters_guessed, warnings, guesses
+
+    is_warning = is_ascii(user_guess) and user_guess in letters  # checks if input a valid letter
+    if is_warning:  # if user's guess was correct
+        if user_guess in my_word:
+            word_letters_guessed.append(user_guess)
+        letters_guessed.append(user_guess)
+        letters = get_available_letters(letters_guessed)
+    else:  # if user's guess was incorrect
+        warnings -= 1
+        if warnings >= 0:
+            print(f'Oops! That is not a valid symbol or you already entered that letter. '
+                  f'You have {warnings} warnings left: ', get_guessed_word(my_word, letters_guessed))
+        else:
+            guesses -= 1
+    return is_warning
+
+
+def get_guesses(user_guess, my_word):
+    """
+    user_guess: input the user is trying to guess;
+    my_word: string with _ characters, current guess of secret word;
+    returns: boolean - True, prints information message
+    and calls func get_guessed_word, if user_guess in word,
+    otherwise False, information message and take away attempts (guesses);
+    """
+    global letters_guessed, guesses
+
+    is_guessed = user_guess in my_word
+    if is_guessed:  # if user guessed a letter
+        print('Good guess:', get_guessed_word(my_word, letters_guessed))
+    else:  # if user didn't guess a letter
+        guesses -= 2 if user_guess in VOWELS else 1
+        if guesses >= 0:
+            print(f'Oops! That letter is not in my word.\n'
+                  f'Please guess a letter: {get_guessed_word(my_word, letters_guessed)}')
+    return is_guessed
+
+
+print('Welcome to the game Hangman!')
+print("I'm thinking of a word that is", len(word), "letters long.\nYou have ", warnings, "warnings and", guesses,
+      "guesses left")
+
+
+def hangman(secret_word, game_mode):
+    """
+    secret_word: string, the secret word to guess.
+    game_mode: 1 - without hints, 2 - with hints
+    returns:
+    Starts up an interactive game of Hangman.
+
+    """
+    global warnings, guesses, letters, letters_guessed
+    if 0 or guesses <= 0:  # if the user has run out of guesses
+        return print('-' * 20, f'\nSorry, you ran out of guesses/warnings. The word was else:\n'
+                               f'"{secret_word}"')
+
+    print('-' * 20, f'\nYou have {guesses} guesses left.\nAvailable letters: {letters}')
+    user_guess = str(input('Please guess a letter: ')).lower()
+
+    if user_guess == HINT and game_mode:  # if the user has input "*" symbol
+        random_words = show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+        print('Possible word matches are:', ', '.join(random_words))  # show 10 possible matches
```

Comment:
Try to have this print inside of function `show_possible_matches`, and try to have identical output as it stated in the task examples